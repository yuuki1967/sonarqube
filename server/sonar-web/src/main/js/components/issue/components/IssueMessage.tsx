/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import EllipsisIcon from '../../icons-components/EllipsisIcon';
import Tooltip from '../../controls/Tooltip';
import { ButtonLink } from '../../ui/buttons';
import { WorkspaceContextShape } from '../../workspace/context';
import { translate, translateWithParameters } from '../../../helpers/l10n';

interface Props {
  engine?: string;
  manualVulnerability: boolean;
  message: string;
  openRule: WorkspaceContextShape['openRule'];
  organization: string;
  rule: string;
}

export default class IssueMessage extends React.PureComponent<Props> {
  handleClick = () => {
    this.props.openRule({
      key: this.props.rule,
      organization: this.props.organization
    });
  };

  render() {
    return (
      <div className="issue-message">
        {this.props.message}
        <ButtonLink
          aria-label={translate('issue.rule_details')}
          className="issue-rule little-spacer-left"
          onClick={this.handleClick}>
          <EllipsisIcon />
        </ButtonLink>
        {this.props.engine && (
          <Tooltip
            overlay={translateWithParameters('issue.from_external_rule_engine', this.props.engine)}>
            <div className="outline-badge badge-tiny-height spacer-left vertical-text-top">
              {this.props.engine}
            </div>
          </Tooltip>
        )}
        {this.props.manualVulnerability && (
          <Tooltip overlay={translate('issue.manual_vulnerability.description')}>
            <div className="outline-badge badge-tiny-height spacer-left vertical-text-top">
              {translate('issue.manual_vulnerability')}
            </div>
          </Tooltip>
        )}
      </div>
    );
  }
}
