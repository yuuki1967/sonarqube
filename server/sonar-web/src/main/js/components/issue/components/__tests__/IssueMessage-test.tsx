/*
 * SonarQube
 * Copyright (C) 2009-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import IssueMessage from '../IssueMessage';

it('should render with the message and a link to open the rule', () => {
  const element = shallow(
    <IssueMessage
      manualVulnerability={false}
      message="Reduce the number of conditional operators (4) used in the expression"
      openRule={jest.fn()}
      organization="myorg"
      rule="javascript:S1067"
    />
  );
  expect(element).toMatchSnapshot();
});
