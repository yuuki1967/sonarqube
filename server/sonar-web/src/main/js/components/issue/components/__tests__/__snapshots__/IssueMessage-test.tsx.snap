// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with the message and a link to open the rule 1`] = `
<div
  className="issue-message"
>
  Reduce the number of conditional operators (4) used in the expression
  <ButtonLink
    aria-label="issue.rule_details"
    className="issue-rule little-spacer-left"
    onClick={[Function]}
  >
    <EllipsisIcon />
  </ButtonLink>
</div>
`;
